### PR TITLE
Only set random seed on purpose in advi

### PR DESCRIPTION
**Pull request by ColCarroll**
Fixes #1395:
The random seeds being set here actually have a different, also probably unexpected side effect: the default argument was always passed to the random number generator, so a call to `check_advi` without setting a random seed will always use the same random number generator. This means that calling the function once, then calling `np.random.random()` will always result in the same number, unless you pass `np.random.get_state()` to the `random_seed` argument.  This diff fixes that to only set a random seed if asked.  

Another thing to note is that the default argument here is `None` rather than `-1`, since -1 is a valid argument to `MRG_RandomStreams` (but not to `np.random.set_seed`).

Apologies for the ugly diff -- my editor removes trailing whitespace.  I can mess with undoing that to make the diff cleaner if you'd like.

### CODE REVIEW

File: pymc3/variational/advi.py [hunk 1]
```diff
@@ -303,7 +306,10 @@ def sample_vp(
     global_RVs = list(set(model.free_RVs) - set(rvs(local_RVs)))
 
     # Make dict for replacements of random variables
-    r = MRG_RandomStreams(seed=random_seed)
+    if random_seed is None:
+        r = MRG_RandomStreams()
```
Comment (taku-y) [inline pymc3/variational/advi.py]: The default seed of MRG_RandomState() is a constant so its result will be the same regardless of the global random state of numpy. Do you intend such behavior if we do not set the seed exlicitly?

Comment (twiecki) [reply pymc3/variational/advi.py]: Yes, I realized the same thing.

We probably have to consider two use-cases:
1. I don't care about the seed and want a different one each time.
2. I want reproducibility.

For 1, could we just copy current `numpy`'s seed value? For 2 we have the kwarg.

Comment (ColCarroll) [reply pymc3/variational/advi.py]: Oh, that's interesting -- I'm running [a test script](https://gist.github.com/ColCarroll/1f54f917cbe50ff9cd792e4af0f921ce), and it looks like the theano random number generator getting a seed doesn't effect the numpy random number generator, so you're right that we just need some way to get a different seed each time.

You can't actually [recover the numpy seed](http://stackoverflow.com/questions/32172054/how-can-i-retrieve-the-current-seed-of-numpys-random-number-generator), since there are too many states to encode in an int64.  I'll try (uniformly) randomly generating a seed using `numpy.random.randint`...

File: pymc3/variational/advi.py [hunk 2]
```diff
@@ -25,39 +20,46 @@
...
+def gen_random_state():
+    """Helper to generate a random state for MRG_RandomStreams"""
+    M1 = 2147483647
+    M2 = 2147462579
+    return np.random.randint(0, M1, 3).tolist() + np.random.randint(0, M2, 3).tolist()
+
+
 def advi(vars=None, start=None, model=None, n=5000, accurate_elbo=False,
-         optimizer=None, learning_rate=.001, epsilon=.1, random_seed=20090425,
+         optimizer=None, learning_rate=.001, epsilon=.1, random_seed=None,
```
Comment (twiecki) [inline pymc3/variational/advi.py:36]: In the other PR I think we were using `-1`, is there a reason to not have consistency?

**Comment by twiecki** [comment]
I assigned @taku-y to review and merge since he's most familiar with that part of the code.

File: pymc3/variational/advi.py (hunk 2 above)
Comment (ColCarroll) [reply pymc3/variational/advi.py:36]: mentioned above -- `-1` is a valid random seed for this generator, but not for numpy.random.seed.  The opposite is true for `None`.  I believe there's an issue in theano to bring their random number generator API in line with numpy's.

Comment (twiecki) [reply pymc3/variational/advi.py:36]: Got it.

**Comment by taku-y** [comment]
It looks good to me. 

@twiecki Should I "Create a merge commit" or "Rebase and merge"?

**Comment by twiecki** [comment]
@taku-y thanks for reviewing. "Rebase and merge".

**Comment by taku-y** [comment]
OK, thanks!
